### DISCUSSION TITLE
Is there a hook or event handler to be able to respond to Chrome Browser crashes?

### DISCUSSION THREAD

**Discussion by samtsai** [cypress-io/cypress#28842] [Questions and Help]
The message below is helpful to know that Chrome renderer crashed, but I'm wondering if there's a way to expose the crash to our tests so that we can run some tasks against the app.

```
We detected that the Chrome Renderer process just crashed.

We have failed the current spec but will continue running the next spec.
This can happen for a number of different reasons.
If you're running lots of tests on a memory intense application.
- Try increasing the CPU/memory on the machine you're running on.
- Try enabling experimentalMemoryManagement in your config file.
- Try lowering numTestsKeptInMemory in your config file during 'cypress open'.

You can learn more here:
https://on.cypress.io/renderer-process-crashed
```

**Comment by portyu9** [comment]
Hi @samtsai 

There isn't a documented Cypress event specifically for **"Chrome renderer crashed"** that you can handle from test code.

More importantly, once the renderer has crashed, code running in that renderer cannot reliably execute another Cypress command such as:

```js
cy.task(...)
```

For cleanup or diagnostics, move the work into the **Node process** and use a lifecycle event such as `after:spec`.

For example, put the actual diagnostic logic in a normal Node function and expose it both as a task and from `after:spec`:

```js
const inspectApplication = async () => {
  // query your app, collect logs, clean up processes, etc.
}

module.exports = {
  e2e: {
    setupNodeEvents(on, config) {
      on('task', {
        inspectApplication,
      })

      on('after:spec', async (spec, results) => {
        if (results?.stats?.failures || results?.error) {
          await inspectApplication()
        }
      })
    },
  },
}
```

`after:spec` runs in Node, so you **cannot call `cy.task()` from it**. Call the underlying Node function directly instead.

One limitation: Cypress does not document a stable `results` field that uniquely means "renderer crash", so the example above reacts to a failed/erroring spec generally rather than claiming to identify only browser crashes.

Also, if the entire Cypress process terminates rather than just the browser renderer, no Cypress lifecycle hook is guaranteed to execute. For that case, diagnostics/cleanup need to live in the process or CI wrapper that launches Cypress.

Please mark this answer as accepted if it helped, thank you.
